### ISSUE TITLE
TS4114 error on new generated commands

### ISSUE THREAD

**Issue by joshcanhelp** [oclif/oclif#1354]
**Is this a bug with the oclif CLI? Or the oclif framework?**
oclif CLI

**Describe the bug**
When generating a new command, I get a TS4114 error telling me that the static properties in the new command class need to indicate that they are an override. I was going to put through a PR for [the template](https://github.com/oclif/oclif/blob/main/templates/src/command.ts.ejs) but figured it would be best to start a discussion here first. 

**To Reproduce**

1. Install `@oclif/core` in a project
2. Run `npm exec oclif generate command hello`
3. Build TypeScript with `tsc`
4. See 4 errors:

```bash
src/commands/hello.ts:4:10 - error TS4114: This member must have an 'override' modifier because it overrides a member in the base class 'Command'.

4   static args = {
           ~~~~

src/commands/hello.ts:8:10 - error TS4114: This member must have an 'override' modifier because it overrides a member in the base class 'Command'.

8   static description = "describe the command here";
           ~~~~~~~~~~~

src/commands/hello.ts:10:10 - error TS4114: This member must have an 'override' modifier because it overrides a member in the base class 'Command'.

10   static examples = ["<%= config.bin %> <%= command.id %>"];
            ~~~~~~~~

src/commands/hello.ts:12:10 - error TS4114: This member must have an 'override' modifier because it overrides a member in the base class 'Command'.

12   static flags = {
```

**Expected behavior**
No TS compile errors

**Your Environment (please complete the following information):**

```bash
$ npm exec oclif -- --version
oclif/4.7.5 darwin-arm64 node-v18.19.1
```

```bash
$ sw_vers
ProductName:            macOS
ProductVersion:         14.3
BuildVersion:           23D56
```

```
$ zsh --version
zsh 5.9 (arm-apple-darwin23.0.0)
```

**Comment by mdonnalley** [comment]
@joshcanhelp Here's what I tried based on the steps you provided:

```
mkdir gh-1354
cd gh-1354
npm init --yes
npm install @oclif/core
npm install typescript --save-dev
npm exec oclif generate command hello
npm exec tsc src/commands/hello.ts
```

I don't see any compilation errors following those steps. Are you able to share a repository that replicates your issue? Thanks!

**Comment by mdonnalley** [comment]
Thanks @joshcanhelp - super helpful

You can get rid of the compilation error by removing `noImplicitOverride` or setting it to `false` in your tsconfig. You could also just add  the `override` modifier to those statics.

Either way, I'm not really sure what we can do on the oclif side to fix this. It would be impractical to have a template (or variety of templates) that will work out of the box for every single tsconfig out there. Based on that, I'm going to close this issue but feel free to keep the conversation going if you think there's something that we could realistically address.

**Comment by joshcanhelp** [comment]
It seems to me like adding `override` in the template is the way to go, not having folks change their tsconfig just for this package. Adding that modifier would not change the behavior for folks that have a less strict config but it would help for those that do. 

Is there a reason why you don't want to add the modifier?

**Comment by mdonnalley** [comment]
>It seems to me like adding override in the template is the way to go

Yeah, that would be a pretty simple and harmless change. I can't imagine most people's tsconfig complaining about the `override` being there even when it's not required.

The larger issue here is that we can't modify the template to fit every person's needs. For instance, someone once posted an issue similar to yours and it turned out that they had turned on every single possible setting in their tsconfig. So there's always going to be some percentage of people for whom the template may not work. 

But like I said, I think adding `override` will be okay for most people. I'll reopen this and then, if you'd like, you can submit a PR to make that change. Otherwise, this might sit on the backlog for a while

**Comment by joshcanhelp** [comment]
> The larger issue here is that we can't modify the template to fit every person's needs.

Totally understand the approach here. It was a simple fix on my end to get it working as needed but who knows if this will stop someone from using oclif in the future if they have less of an understanding of what's going on here.
